src/Compilers/VisualBasic/Portable/Lowering/SyntheticBoundNodeFactory.vb

```diff
@@ -269,6 +269,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 useSiteInfo = Binder.GetUseSiteInfoForMemberAndContainingType(memberSymbol)
             End If
 
+            If useSiteInfo.DiagnosticInfo IsNot Nothing AndAlso isOptional Then
+                Return Nothing
```

Comment:
Do we want to distinguish between warnings and errors here?

Reply:
> Do we want to distinguish between warnings and errors here?

We don't need to. There are no use-site warnings in VB. For example, see logic in the method above.